server/src/main/kotlin/io/provenance/objectstore/gateway/service/JwtVerificationService.kt

```diff
@@ -0,0 +1,55 @@
+package io.provenance.objectstore.gateway.service
+
+import com.auth0.jwt.JWT
+import com.auth0.jwt.algorithms.Algorithm
+import com.auth0.jwt.exceptions.JWTVerificationException
+import com.auth0.jwt.interfaces.DecodedJWT
+import org.springframework.stereotype.Service
+import java.security.PublicKey
+import io.provenance.objectstore.gateway.exception.JwtValidationException
+import io.provenance.scope.encryption.ecies.ECUtils
+import io.provenance.scope.encryption.util.getAddress
+import io.provenance.scope.objectstore.util.base64Decode
+import java.security.interfaces.ECPublicKey
+
+data class VerificationResult(val address: String, val publicKey: PublicKey)
+
+@Service
+class JwtVerificationService {
+    fun verifyJwtString(jwt: String) = verifyJwt(JWT.decode(jwt))
+
+    fun verifyJwt(jwt: DecodedJWT): VerificationResult {
+        val publicKey = jwt.claims.get(Constants.JWT_PUBLIC_KEY)
+            ?.asString()
+            ?.base64Decode()
+            ?.let { ECUtils.convertBytesToPublicKey(it) }
+            ?: throw JwtValidationException("Failed to find public key in jwt")
+
+        val algorithm = Algorithm.ECDSA256K(publicKey as ECPublicKey, null)
+
+        val verified = try {
+            JWT.require(algorithm)
+                .acceptExpiresAt(0L)
+                .build()
+                .verify(jwt)
+        } catch (e: JWTVerificationException) {
+            throw JwtValidationException(e.message)
+        }
+
+        if (!verified.claims.containsKey(Constants.JWT_EXPIRATION_KEY)) {
```

Comment:
Does this blow up above if the expiration time has already passed?  If not, might need to add a check in here that OffsetDateTime.now() <= expiration time.  Likely will include timezone screwery, though lol

Reply:
Ah, I see the test that proves that this is a thing.  Don't mind me :disappear:

Reply:
yeah, apparently the expiration is optional and if missing the verification doesn't care, and there didn't seem to be an option to force it to care. Seemed important that we don't allow non-expiring jwts though